packages/api/resolvers/mutations/updateCart.js

```diff
@@ -15,21 +15,13 @@ export default async function updateCart(
 ) {
   log('mutation updateCart', { userId });
 
-  let order = await getCart({ orderId, user, countryContext });
-  if (meta) {
-    order = order.updateContext(meta);
-  }
-  if (billingAddress) {
-    order = order.updateBillingAddress(billingAddress);
-  }
-  if (contact) {
-    order = order.updateContact(contact);
-  }
-  if (paymentProviderId) {
-    order = order.setPaymentProvider({ paymentProviderId });
-  }
-  if (deliveryProviderId) {
-    order = order.setDeliveryProvider({ deliveryProviderId });
-  }
-  return order;
+  const order = await getCart({ orderId, user, countryContext });
+
+  return order.updateCart({
```

Comment:
I think it's better to revert moving this to an own updateCart helper because it's abstracted good enough already. It can be seen as the decision of the API designer to allow changing multiple aspects of an order with one updateCart mutation but it's not really core.